File: src/main/java/dev/retrotv/random/generator/PINGenerator.java
package dev.retrotv.random.generator;

import java.util.Random;

import static dev.retrotv.random.enums.SecurityStrength.ONLY_NUMBER;

/**
 * 무작위 PIN 생성을 위한 클래스 입니다.
 *
 * @author yjj8353
 * @since 2.0.0
 */
public class PINGenerator extends StringGenerator {

    /**
     * 기본 생성자
     * 기본 생성 길이는 4자리로 별도 설정됩니다.
     *
     * @author  yjj8353
     * @since   2.0.0
     * @param random Random 객체
     */
    PINGenerator(Random random) {
        super(random, ONLY_NUMBER);
        this.setLength(4);
    }
}

File: src/main/kotlin/dev/retrotv/random/PINGenerator.kt
package dev.retrotv.random

import dev.retrotv.random.enums.SecurityStrength
import java.util.Random

/**
 * 무작위 PIN 생성을 위한 클래스 입니다.
 *
 * @author yjj8353
 * @since 1.0.0
 * @property random 무작위 데이터 생성을 위한 Random 객체
 */
class PINGenerator(random: Random)
    : StringGenerator(SecurityStrength.ONLY_NUMBER, random)
